src/commands/createNewProject/IProjectCreator.ts

```diff
@@ -3,21 +3,46 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { OutputChannel } from "vscode";
+import { TelemetryProperties } from "vscode-azureextensionui";
+import { IUserInterface } from "../../IUserInterface";
 import { localize } from "../../localize";
-import { extensionPrefix, ProjectRuntime, TemplateFilter } from "../../ProjectSettings";
+import { extensionPrefix, ProjectRuntime, promptForProjectRuntime, TemplateFilter } from "../../ProjectSettings";
+import { functionRuntimeUtils } from "../../utils/functionRuntimeUtils";
 
-export interface IProjectCreator {
-    deploySubpath?: string;
-    runtime: ProjectRuntime;
-    templateFilter: TemplateFilter;
+export abstract class ProjectCreatorBase {
```

Comment:
Should change the name of the file as this is no longer an interface

Reply:
Sounds good. Just filed an issue to address separately: https://github.com/Microsoft/vscode-azurefunctions/issues/251

This isn't the only file I want to rename and I'd prefer to do it separately